network/p2p/scoring/subscription_validator.go

```diff
@@ -0,0 +1,49 @@
+package scoring
+
+import (
+	"github.com/libp2p/go-libp2p/core/peer"
+
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/network/p2p"
+	p2putils "github.com/onflow/flow-go/network/p2p/utils"
+)
+
+type SubscriptionValidator struct {
+	idProvider           module.IdentityProvider
+	subscriptionProvider p2p.SubscriptionProvider
+}
+
+func NewSubscriptionValidator(idProvider module.IdentityProvider) *SubscriptionValidator {
+	return &SubscriptionValidator{
+		idProvider: idProvider,
+	}
+}
+
+func (v *SubscriptionValidator) RegisterSubscriptionProvider(provider p2p.SubscriptionProvider) {
+	v.subscriptionProvider = provider
+}
+
+// CheckSubscribedToAllowedTopics validates all subscriptions a peer has with respect to all Flow topics.
+// All errors returned by this method are benign:
+// We expect an unauthorized or ejected peer to subscribe to Flow topics (InvalidPeerIDError).
+// We expect a peer to subscribe to a topic that is not allowed for its role (InvalidSubscriptionError).
```

Comment:
I find these a little confusing. maybe rephrase to something like this?
```suggestion
// - InvalidPeerIDError: the subscribed peer is unauthorized (unknown) or ejected.
// - InvalidSubscriptionError: the peer is subscribed to a topic that is not allowed for its role.
```